tests/python_package_test/test_engine.py

```diff
@@ -428,6 +428,72 @@ def test_multi_class_error(self):
                   valid_sets=[lgb_data], evals_result=results, verbose_eval=False)
         self.assertAlmostEqual(results['training']['multi_error@2'][-1], 0)
 
+    def test_auc_mu(self):
+        # should give same result as binary auc for 2 classes
+        X, y = load_digits(10, True)
+        y_new = np.zeros((len(y)))
+        y_new[y != 0] = 1
+        lgb_X = lgb.Dataset(X, label=y_new)
+        params = {'objective': 'multiclass',
+                  'metric': 'auc_mu',
+                  'verbose': -1,
+                  'num_classes': 2,
+                  'seed': 0}
+        results_auc_mu = {}
+        lgb.train(params, lgb_X, num_boost_round=10, valid_sets=[lgb_X], evals_result=results_auc_mu)
+        lgb_X = lgb.Dataset(X, label=(y == 0).astype(int))
+        params = {'objective': 'binary',
+                  'metric': 'auc',
+                  'verbose': -1,
+                  'seed': 0}
+        results_auc = {}
+        lgb.train(params, lgb_X, num_boost_round=10, valid_sets=[lgb_X], evals_result=results_auc)
+        np.testing.assert_allclose(results_auc_mu['training']['auc_mu'], results_auc['training']['auc'])
+        # test the case where all predictions are equal
+        lgb_X = lgb.Dataset(X[:10], label=y_new[:10])
+        params = {'objective': 'multiclass',
+                  'metric': 'auc_mu',
+                  'verbose': -1,
+                  'num_classes': 2,
+                  'min_data_in_leaf': 20,
+                  'seed': 0}
+        results_auc_mu = {}
+        lgb.train(params, lgb_X, num_boost_round=10, valid_sets=[lgb_X], evals_result=results_auc_mu)
+        self.assertAlmostEqual(results_auc_mu['training']['auc_mu'][-1], 0.5)
+        # should give 1 when accuracy = 1
+        X, y = load_digits(10, True)
```

Comment:
Seems that `X` and `y` remained untouched before this line, so there is no need to load data once again.